src/overlays/actors/ovl_En_Pamera/z_en_pamera.c

```diff
@@ -21,120 +103,808 @@ const ActorInit En_Pamera_InitVars = {
+            Actor_MarkForDeath(&this->actor);
+        }
+        if ((gSaveContext.weekEventReg[61] & 4) != 0) {
+            if (((gSaveContext.weekEventReg[59] & 1) == 0) || (gSaveContext.entranceIndex != 0x2020)) {
+                if ((gSaveContext.entranceIndex != 0x2020) && ((gSaveContext.weekEventReg[59] & 1) != 0)) {
+                    gSaveContext.weekEventReg[59] &= 0xFE;
+                }
+                func_80BD8700(this);
+            } else {
+                this->pathIndex = this->pathPointsCount - 1;
+                Math_Vec3s_ToVec3f(&this->actor.world.pos, &this->pathPoints[this->pathIndex]);
+                Math_Vec3s_ToVec3f(&sp44, &this->pathPoints[this->pathIndex - 1]);
+                this->actor.world.rot.y = this->actor.shape.rot.y = Math_Vec3f_Yaw(&sp44, &this->actor.world.pos);
+                func_80BD8CCC(this);
+            }
+        } else {
+            gSaveContext.weekEventReg[59] |= 1;
+            func_80BD8FF0(this);
+        }
+    }
+}
+
+s32 func_80BD84F0(EnPamera* this, GlobalContext* globalCtx) {
+    Actor* actorPtr = globalCtx->actorCtx.actorList[ACTORCAT_DOOR].first;
+
+    while (actorPtr != 0) {
+        if ((actorPtr->id == ACTOR_EN_DOOR) && (Math_Vec3f_DistXZ(&this->actor.world.pos, &actorPtr->world.pos) < 200.0f)) {
+            this->actor.child = actorPtr;
+            return 1;
+        }
+        actorPtr = actorPtr->next;
+    }
+    this->actor.child = NULL;
+    return 0;
+}
+
+void func_80BD8588(EnPamera* this, GlobalContext* globalCtx) {
+    Path* path = &globalCtx->setupPathList[((this->actor.params & 0xFF0) >> 4)];
+    Vec3f sp28;
+
+    if (path == 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    if (gSaveContext.weekEventReg[61] & 4) {
+        path = &globalCtx->setupPathList[(u8)path->unk1];
+    }
+    this->pathPoints = Lib_SegmentedToVirtual(path->points);
+    this->pathIndex = 0;
+    this->pathPointsCount = path->count;
+    this->animatedTextures = (u8)path->unk1;
+    Math_Vec3s_ToVec3f(&sp28, this->pathPoints);
+    this->actor.world.rot.y = this->actor.shape.rot.y = Math_Vec3f_Yaw(&this->actor.world.pos, &sp28);
+}
+
+void func_80BD8658(EnPamera* this) {
+    s16 nextCutscene;
+    s32 i;
+
+    nextCutscene = this->actor.cutscene;
+    for (i = 0; i < 2; i++) {
+        if (nextCutscene == -1) {
+            break;
+        }
+        this->cutscenes[i] = nextCutscene;
+        nextCutscene = ActorCutscene_GetAdditionalCutscene(nextCutscene);
+    }
+}
+
+void EnPamera_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnPamera* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80BD8700(EnPamera* this) {
+    this->unk_31C = 0;
+    this->actor.flags &= -2;
+    func_800BDC5C(&this->skelAnime, animations, 0);
+    this->actionFunc = func_80BD8758;
+}
+
+void func_80BD8758(EnPamera* this, GlobalContext* globalCtx) {
+    s32 temp = (this->unk_31C < 0x709) ^ 1;
+
+    this->unk_31C++;
+    if (temp) {
```

Comment:
```suggestion
    if (this->unk_31C++ >= 0x709) {
```
I was curious so I checked out the PR, and it looks like this matches. It probably applies to all the other situations like this.